packages/mimecast/data_stream/archive_search_logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,155 @@
+---
+description: Pipeline for processing Mimecast archive_search_logs.
+processors:
+  # Generic event/ecs fields we always want to populate.
+  - set:
+      field: ecs.version
+      value: '8.9.0'
+  - rename:
+      field: message
+      target_field: event.original
+  - json:
+      description: Parse 'message' JSON contents into a 'mimecast' object.
+      field: event.original
+      target_field: mimecast
+  - drop:
+      if: ctx.mimecast?.createTime == null || (ctx.mimecast?.data != null && ctx.mimecast.data.isEmpty())
+  - fingerprint:
+      fields:
+        - mimecast.id
+      target_field: "_id"
+      ignore_missing: true
+
+  - date:
+      description: Use 'mimecast.createTime' as the '@timestamp'
+      field: mimecast.createTime
+      timezone: UTC
+      formats:
+        - "yyyy-MM-dd'T'HH:mm:ssz"
+        - "yyyy-MM-dd'T'HH:mm:ssZ"
+        - "yyyy-MM-dd'T'HH:mm:ss.Sz"
+        - "yyyy-MM-dd'T'HH:mm:ss.SZ"
+        - "yyyy-MM-dd'T'HH:mm:ss.SSz"
+        - "yyyy-MM-dd'T'HH:mm:ss.SSZ"
+        - "yyyy-MM-dd'T'HH:mm:ss.SSSz"
+        - "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        - "yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSz"
+        - "yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSZ"
+        - "yyyy-MM-dd'T'HH:mm:ss z"
+
+  ###
+
+  # Search related fields
+  - rename:
+      field: mimecast.searchText
+      target_field: mimecast.search_details.text
+      ignore_missing: true
+  - rename:
+      field: mimecast.description
+      target_field: mimecast.search_details.description
+      ignore_missing: true
+  - rename:
+      field: mimecast.source
+      target_field: mimecast.search_details.source
+      ignore_missing: true
+  - rename:
+      field: mimecast.searchPath
+      target_field: mimecast.search_details.path
+      ignore_missing: true
+  - rename:
+      field: mimecast.searchReason
+      target_field: mimecast.search_details.reason
+      ignore_missing: true
+
+  ###
+
+  # User fields
+  - rename:
+      field: mimecast.emailAddr
+      target_field: user.email
+      ignore_missing: true
+
+  - split:
+      field: user.email
+      separator: "@"
+      target_field: user.parts
+      if: 'ctx.user?.email != null'
+  - set:
+      field: user.name
+      copy_from: user.parts.0
+      if: 'ctx.user?.parts?.length > 1'
```

```suggestion
      if: 'ctx.user?.parts != null && ctx.user.parts.length > 1'
```

I think this has to be first checked for `null`. Same below

---

Doesn't `parts?.` do a null check on this?

---

🤔 I never tried that. Maybe you can run a quick test with `user.parts` null. Maybe it works.

---

Apparently, `user.parts` is never null. If `mimecast.emailAddr` is not present then `user.email` is null and hence split shall not run on it at all. 

And if `split` does not find `separator` then the `field` is returned as is into `target_field`.

---

So does that mean we could say this also as `ctx.user.parts instanceof Array` or `ctx.user.parts.length != null`?

---

A `split` processor ( if not eun against `null` ) shall return an array with one element atleast ( even if it cannot find the separator ). 

So, if it is an array, length always is a value >= 1

---

OK, so not, "as is". Thanks.